src/sagemaker/session.py

```diff
@@ -4114,6 +4126,21 @@ def submit(request):
                         ModelPackageGroupName=request["ModelPackageGroupName"]
                     )
                 )
+            if "SourceUri" in request and request["SourceUri"] is not None:
+                if re.match(MODEL_PACKAGE_ARN_PATTERN, request["SourceUri"]) or re.match(
+                    MODEL_ARN_PATTERN, request["SourceUri"]
+                ):
+                    if "InferenceSpecification" in request:
+                        del request["InferenceSpecification"]
+                    return self.sagemaker_client.create_model_package(**request)
+                request_source_uri = request["SourceUri"]
+                del request["SourceUri"]
+                model_package = self.sagemaker_client.create_model_package(**request)
+                update_source_uri_args = {
+                    "ModelPackageArn": model_package.get("ModelPackageArn"),
+                    "SourceUri": request_source_uri,
+                }
+                return self.sagemaker_client.update_model_package(**update_source_uri_args)
```

Comment:
Can you add comment in code why doing create and then update? I myself bit confused. I believe it is for the case where model(inference spec is present) we would first go with inference spec during create and then add sourceUri